front/components/agent_builder/instructions/InstructionBlockExtension.tsx

```diff
@@ -0,0 +1,246 @@
+import { InputRule, mergeAttributes, Node } from "@tiptap/core";
+import type { Node as ProseMirrorNode } from "@tiptap/pm/model";
+import type { Slice } from "@tiptap/pm/model";
+import { Plugin, PluginKey } from "@tiptap/pm/state";
+import { TextSelection } from "@tiptap/pm/state";
+import type { EditorView } from "@tiptap/pm/view";
+import type { NodeViewProps } from "@tiptap/react";
+import {
+  NodeViewContent,
+  NodeViewWrapper,
+  ReactNodeViewRenderer,
+} from "@tiptap/react";
+import React from "react";
+
+import {
+  createProseMirrorInstructionBlock,
+  INSTRUCTION_BLOCK_REGEX,
+  OPENING_TAG_REGEX,
+  parseInstructionBlockMatches,
+  splitTextAroundBlocks,
+} from "@app/lib/client/assistant_builder/instructionBlockUtils";
+
+export interface InstructionBlockAttributes {
+  type: string;
+}
+
+declare module "@tiptap/core" {
+  interface Commands<ReturnType> {
+    instructionBlock: {
+      setInstructionBlock: (
+        attributes: InstructionBlockAttributes
+      ) => ReturnType;
+      insertInstructionBlock: (type: string) => ReturnType;
+    };
+  }
+}
+
+const InstructionBlockComponent: React.FC<NodeViewProps> = ({ node }) => {
+  const { type } = node.attrs as InstructionBlockAttributes;
+
+  return (
+    <NodeViewWrapper className="my-4">
+      <div className="rounded-lg border border-border bg-muted-background p-4">
+        <div className="mb-3">
+          <span
+            className="text-sm font-semibold text-muted-foreground"
+            contentEditable={false}
+          >
+            {type.toUpperCase()}
+          </span>
+        </div>
+        <div className="min-h-12 rounded border border-border/50 bg-background p-3">
+          <NodeViewContent className="prose prose-sm" />
+        </div>
+      </div>
+    </NodeViewWrapper>
+  );
+};
+
+export const InstructionBlockExtension =
```

Comment:
Do you know from where we are inserting this trailing break? 🤔 It's added every time you add instruction block:

<img width="720" height="302" alt="Image" src="https://github.com/user-attachments/assets/05c593ef-d862-4b56-8cf1-6fdd3a624ce6" />

Reply:
I think we can add trailing break after the instruction block, this way probably you can get out from the block by clicking there? (right now there is no way to go to the outside/bottom of the block except pressing enter twice)

Reply:
Definitely (will do later, couldn't figure out easily)